From 8faf428752c69acb6acaf16957106043feb494c9 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Wed, 3 Nov 2021 16:40:18 +0100
Subject: [PATCH 0160/1695] clk: add is_enabled() ops

Sometimes we need to know the real status of a clock, after bl2 boot or
after a cstop for example.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 20f933092c2e545ca6fa98c3d460aaa2dfed5ff1 clk: add is_enabled() ops]
Change-Id: If0555af852a7b357aa72ff4dae454e3a92629e92
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
---
 core/include/drivers/clk.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/include/drivers/clk.h b/core/include/drivers/clk.h
index 20b7d5405..a6ce2484b 100644
--- a/core/include/drivers/clk.h
+++ b/core/include/drivers/clk.h
@@ -47,6 +47,7 @@ struct clk {
 /**
  * struct clk_ops
  *
+ * @is_enabled: Get effective state of the clock (on / off)
  * @enable: Enable the clock
  * @disable: Disable the clock
  * @set_parent: Set the clock parent based on index
@@ -56,6 +57,7 @@ struct clk {
  * @get_rates_array: Get the supported clock rates as array
  */
 struct clk_ops {
+	bool (*is_enabled)(struct clk *clk);
 	TEE_Result (*enable)(struct clk *clk);
 	void (*disable)(struct clk *clk);
 	TEE_Result (*set_parent)(struct clk *clk, size_t index);
-- 
2.47.2

